cmd/gctcli/commands.go

```diff
@@ -2148,45 +2175,481 @@ func getCryptocurrencyDepositAddress(c *cli.Context) error {
+			Flags: []cli.Flag{
+				cli.StringFlag{
+					Name:  "exchange",
+					Usage: "<exchange>",
+				},
+				cli.StringFlag{
+					Name:  "start",
+					Usage: "<start>",
+				},
+				cli.StringFlag{
+					Name:  "end",
+					Usage: "<end>",
+				},
+				cli.Int64Flag{
+					Name:  "limit",
+					Usage: "<limit>",
+				},
+			},
+			Action: withdrawlRequestByDate,
+		},
 	},
 }
 
-func withdrawFiatFunds(_ *cli.Context) error {
-	return common.ErrNotYetImplemented
+func withdrawlRequestByID(c *cli.Context) error {
+	if c.NArg() == 0 && c.NumFlags() == 0 {
+		cli.ShowSubcommandHelp(c)
+		return nil
+	}
+
+	var ID string
+	if c.IsSet("id") {
+		ID = c.String("id")
+	} else {
+		ID = c.Args().First()
+	}
+
+	if ID == "" {
+		return errors.New("a ID must be specified")
+	}
+
+	conn, err := setupClient()
+	if err != nil {
+		return err
+	}
+	defer conn.Close()
+
+	client := gctrpc.NewGoCryptoTraderClient(conn)
+
+	result, err := client.WithdrawalEventByID(context.Background(),
+		&gctrpc.WithdrawalEventByIDRequest{
+			Id: ID,
+		},
+	)
+	if err != nil {
+		return err
+	}
+	jsonOutput(result)
+	return nil
+}
+
+func withdrawlRequestByExchangeID(c *cli.Context) error {
+	if c.NArg() == 0 && c.NumFlags() == 0 {
+		cli.ShowSubcommandHelp(c)
+		return nil
+	}
+
+	var exchange string
+	if c.IsSet("exchange") {
+		exchange = c.String("exchange")
+	} else {
+		exchange = c.Args().First()
+	}
+
+	var limit, limitStr int64
+	var ID string
+	var err error
+	if c.Command.Name == "byexchangeid" {
+		if c.IsSet("id") {
+			ID = c.String("id")
+		} else {
+			ID = c.Args().Get(1)
+		}
+		if ID == "" {
+			return errors.New("a ID must be specified")
+		}
+		limit = 1
+	} else {
+		if c.IsSet("limit") {
+			limit = c.Int64("limit")
+		} else if c.Args().Get(1) != "" {
+			limitStr, err = strconv.ParseInt(c.Args().Get(1), 10, 64)
+			if err != nil {
+				return err
+			}
+			if limitStr > math.MaxInt32 {
+				return fmt.Errorf("limit greater than max size: %v", math.MaxInt32)
+			}
+			limit = limitStr
+		}
+	}
+
+	conn, err := setupClient()
+	if err != nil {
+		return err
+	}
+	defer conn.Close()
+
+	client := gctrpc.NewGoCryptoTraderClient(conn)
+
+	result, err := client.WithdrawalEventsByExchange(context.Background(),
+		&gctrpc.WithdrawalEventsByExchangeRequest{
+			Exchange: exchange,
+			Id:       ID,
+			Limit:    int32(limit),
+		},
+	)
+	if err != nil {
+		return err
+	}
+	jsonOutput(result)
+	return nil
+}
+
+func withdrawlRequestByDate(c *cli.Context) error {
+	if c.NArg() == 0 && c.NumFlags() == 0 {
+		cli.ShowSubcommandHelp(c)
+		return nil
+	}
+
+	var exchange, start, end string
+	var limit, limitStr int64
+	var err error
+	if c.IsSet("exchange") {
+		exchange = c.String("exchange")
+	} else {
+		exchange = c.Args().First()
+	}
+
+	if c.IsSet("start") {
+		start = c.String("start")
+	} else {
+		start = c.Args().Get(1)
+	}
+
+	if c.IsSet("end") {
+		end = c.String("end")
+	} else {
+		end = c.Args().Get(2)
```

Comment:
Perhaps `if c.Args().Get(2) != "" { end = x }` and if end is "", can set it to time.Now()

Reply:
Last nit I have is for this, it would be more user friendly to have default vars set for the start and end time periods, it also helps show users what time format is expected (for the start time, can do the past 30 days and for the end time, can use time.Now())

Reply:
Will add it!

Reply:
```sh
 go run ./cmd/gctcli withdrawalrequesthistory bydate
NAME:
   gctcli withdrawalrequesthistory bydate - exchange start end limit

USAGE:
   gctcli withdrawalrequesthistory bydate [command options] <exchange> <start> <end> <limit>

OPTIONS:
   --exchange value  <exchange>
   --start value     <start> (default: "2020-01-21 13:56:53")
   --end value       <end> (default: "2020-02-21 13:56:53")
   --limit value     <limit> (default: 0)
```

Default values have been added